js/pro/gate.js

```diff
@@ -417,6 +448,13 @@ module.exports = class gate extends gateRest {
             const messageHash = channel + '.' + symbol;
             client.resolve (this.tickers[symbol], messageHash);
         }
+        const messageHashes = Object.keys (client.futures);
+        for (let i = 0; i < messageHashes.length; i++) {
+            const messageHash = messageHashes[i];
```

Comment:
@sc0Vu this works for the basic scenario, but won't work if the user does, for instance this:

```
 Promise.all(gate.watchTickers(['BTC/USDT', 'LTC/USDT'], gate.watchTicker("ETH/USDT"))
```
⬆️  you blindly resolve the `tickers` promise, but you're not checking inside `watchTickers` if the resolved symbol is "valid", it might come from the `watchTicker('ETH/USDT')` update

Reply:
Does `filterByArray` filter the symbols from tickers?

Reply:
@sc0Vu it does, but if the symbol is not part of the subscription it will return an empty array to the user, and we want to avoid that